cdap-unit-test/src/test/java/net/fake/test/app/TestBundleJarApp.java

```diff
@@ -44,51 +45,47 @@
 public class TestBundleJarApp extends TestBase {
 
   @Test
-  public void testFlow() throws IOException, URISyntaxException, TimeoutException, InterruptedException {
+  public void testBundleJar() throws IOException, URISyntaxException, TimeoutException, InterruptedException {
     File helloWorldJar = new File(TestBundleJarApp.class.getClassLoader().getResource("helloworld.jar").toURI());
     ApplicationManager applicationManager = deployApplication(BundleJarApp.class, helloWorldJar);
 
-    FlowManager flowManager = applicationManager.startFlow("SimpleFlow");
-    StreamWriter streamWriter = applicationManager.getStreamWriter("simpleInputStream");
-    for (int i = 0; i < 5; i++) {
-      streamWriter.send("test" + i + ":" + i);
-    }
+    try {
+      FlowManager flowManager = applicationManager.startFlow("SimpleFlow");
+      StreamWriter streamWriter = applicationManager.getStreamWriter("simpleInputStream");
+      for (int i = 0; i < 5; i++) {
+        streamWriter.send("test" + i + ":" + i);
+      }
 
-    // Check the flowlet metrics
-    RuntimeMetrics flowletMetrics = RuntimeStats.getFlowletMetrics("BundleJarApp", "SimpleFlow", "SimpleFlowlet");
-    Thread.sleep(3000);
+      // Check the flowlet metrics
+      RuntimeMetrics flowletMetrics = RuntimeStats.getFlowletMetrics("BundleJarApp", "SimpleFlow", "simpleFlowlet");
+      flowletMetrics.waitForProcessed(5, 5, TimeUnit.SECONDS);
+      Assert.assertEquals(0L, flowletMetrics.getException());
+      flowManager.stop();
 
-    // TODO: not working
```

Comment:
Shoot, I guess you fixed it in the lines above. Sometimes git shows diffs out of order... please ignore this